src/plugins/topic_viewer/TopicViewer.qml

```diff
@@ -0,0 +1,252 @@
+        // item pos changes randomly when drag happens (with the copy drag)
+        anchors.top: parent.top
+        anchors.right: parent.right
+
+        Drag.mimeData: { "text/plain" : (model === null) ? "" : model.topic + "," + model.path }
+
+        Drag.dragType: Drag.Automatic
+        Drag.supportedActions : Qt.CopyAction
+        Drag.active: dragMouse.drag.active
+        // a point to drag from
+        Drag.hotSpot.x: 0
+        Drag.hotSpot.y: itemHeight
+
+        // used by DropArea that accepts the dragged items
+        function itemData ()
+        {
+            return {
+                "name": model.name,
+                "type": model.type,
+                "path": model.path,
+                "topic": model.topic
+            }
+        }
+
+        MouseArea {
+            id: dragMouse
+            anchors.fill: parent
+
+            // only plottable items are dragable
+            drag.target: (model === null) ? null :
+                         (model.plottable) ? parent : null
+
+            // get a copy image of the dragged item
+            onPressed: parent.grabToImage(function(result) {
+                parent.Drag.imageSource = result.url
+            })
+
+            onReleased:
+            {
+                // emit drop event to notify the DropArea (must manually)
+                parent.Drag.drop();
+            }
+
+            hoverEnabled: true
+            propagateComposedEvents: true
+            // make the cursor with a drag shape at the plottable items
+            cursorShape: (model === null) ?  Qt.ArrowCursor : (model.plottable) ?
+                                                Qt.DragCopyCursor : Qt.ArrowCursor
+
+            onClicked: {
+                // change the selection of the tree by clearing the prev, select a new one
+                tree.selection.select(styleData.index,ItemSelectionModel.ClearAndSelect)
+
+                // set the selection index to the index of the clicked item (must set manually)
+                tree.selection.setCurrentIndex(styleData.index,ItemSelectionModel.ClearAndSelect)
+
+                // the currentIndex of the tree.selection is not the same
+                // of the tree.currentIndex, so set the tree.currentIndex.
+                // this is the way to access it as it is read-only
+                tree.__currentRow = styleData.row
+
+                // set the focus to the selected item to receive the keyboard events
+                // this is useful to enable navigating with keyboard from the right position
+                item.forceActiveFocus();
+
+                tree.expandCollapseMsg(tree.currentIndex);
+            }
+        }
+
+        Image {
+            id: icon
+            source: "plottable_icon.svg"
+            height: itemHeight * 0.6
+            width: itemHeight * 0.6
+            y : itemHeight * 0.2
+            visible: (model === null) ? false : model.plottable
+        }
+
+        Text {
```

Comment:
The text is leaking when it's larger than the space available on the plugin. If you do this:

```.diff
diff --git a/src/plugins/topic_viewer/TopicViewer.qml b/src/plugins/topic_viewer/TopicViewer.qml
index 93aa675..7aba69e 100644
--- a/src/plugins/topic_viewer/TopicViewer.qml
+++ b/src/plugins/topic_viewer/TopicViewer.qml
@@ -164,12 +164,14 @@ TreeView {
             font.pointSize: 12
             anchors.leftMargin: 5
             anchors.left: icon.right
+            anchors.right: parent.right
             y: icon.y
+            elide: Text.ElideMiddle
         }
 
         ToolTip {
             id: tool_tip
```

Then the text will elide in the middle (with ...) when there isn't enough space:

![elide_topic](https://user-images.githubusercontent.com/5751272/88336917-ad4ade80-ccea-11ea-84ef-fdcbd1cc8bdb.gif)